### PR TITLE
fix: adding new test cases for git auto-commit

### LINKED ISSUE #33964: [Task]: Cypress test cases for autocommit
### SubTasks

Following are the test cases -
- Import an old repo > go to application > check for auto-commit status bar
- Import an old repo > go to application > check for auto-commit trigger and progress api
- Import an old repo > go to application > change branch when auto-commit is done

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Adds cypress test for checking auto-commit functionality

Fixes #33964

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10715693875>
> Commit: 3dda8362544b4bc7b977e780f7d1140b1343c93f
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10715693875&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Git`
> Spec:
> <hr>Thu, 05 Sep 2024 07:45:36 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced an end-to-end test suite for the Git autocommit feature, validating UI and backend interactions.
	- Added a new property to track the autocommit status bar in the GitSync class.
	- Enhanced the AutocommitStatusbar component for better testability with a new identifier.

- **Improvements**
	- Made auto-commit eligibility checks always available by removing feature flag gating.
	- Improved logging visibility for auto-commit operations by adjusting log levels.

- **Bug Fixes**
	- Streamlined tests by removing unnecessary checks related to the auto-commit eligibility feature flag.

### CODE REVIEW

**Comment by rajatagrawal** [comment]
Can we please have react testing library test for this change ?

cc : @ApekshaBhosale @yatinappsmith @sagar-qa007

**Review by rajatagrawal — CHANGES_REQUESTED** [review]
Left comments

**Comment by brayn003** [comment]
> Can we please have react testing library test for this change ?

@rajatagrawal can you please elaborate?

**Comment by brayn003** [comment]
`react-testing-library` is good for testing components in isolation, that is not the use case here.
The intention is to test the flow for auto-commit in it's entirety. Hence, the Cypress approach

**Comment by ApekshaBhosale** [comment]
@brayn003 spec changes LGTM

**Review by ApekshaBhosale — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          ).to.equal("PUBLISHED");
+          agHelper.WaitUntilEleAppear(gitSync._autocommitStatusbar);
+        });
+        cy.wait("@gitAutocommitProgressApi").then((interceptions) => {
+          expect(interceptions?.response?.statusCode).to.equal(200);
+        });
+      });
+    });
+  },
+);
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts:57]: Cy wait here is used with API call aliases as you suggested

File: app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import ReconnectLocators from "../../../../locators/ReconnectLocators";
+import { featureFlagIntercept } from "../../../../support/Objects/FeatureFlags";
+import {
+  agHelper,
+  gitSync,
+  homePage,
+} from "../../../../support/Objects/ObjectsCore";
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts:7]: Reconnect attributes include data-* css selectors in them

File: app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          expect(interception?.response?.statusCode).to.equal(200);
+          expect(
+            interception?.response?.body?.data?.autoCommitResponse,
+          ).to.equal("PUBLISHED");
+          agHelper.WaitUntilEleAppear(gitSync._autocommitStatusbar);
+        });
+        cy.wait("@gitAutocommitProgressApi").then((interceptions) => {
+          expect(interceptions?.response?.statusCode).to.equal(200);
+        });
+      });
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts:54]: Chai TDD APIs don't support this

File: app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          },
+          (req) => {
+            req.on("response", (res) => {
+              res.setDelay(500);
+            });
+          },
+        ).as("gitAutocommitProgressApi");
+
+        gitSync.ImportAppFromGit(wsName, repoName);
+        agHelper.GetNClick(ReconnectLocators.SkipToAppBtn);
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts:43]: This is already a standard practice we follow in all our test cases

File: app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  { tags: ["@tag.Git", "@tag.GitAutocommit"] },
+  function () {
+    it("Check if autocommit progress bar is visible and network requests are properly called", function () {
+      featureFlagIntercept({
+        release_git_autocommit_feature_enabled: true,
+      });
+      agHelper.GenerateUUID();
+      cy.get("@guid").then((uid) => {
+        wsName = "GitAC-" + uid;
+        homePage.CreateNewWorkspace(wsName, true);
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/cypress/e2e/Regression/ClientSide/Git/GitAutocommit_spec.ts:23]: This is already a standard practice we follow in all our test cases

**Pull request merged**